XpressAI/xircuits
# ✨ Enable Arg Parsing in Xircuits Entry Point

# Description

This PR enables passing arguments from the xircuits console entry point. I've applied it so that we can pull `xai_components`, `examples` and `datasets` from branches. 

The two things you can do now is:

1. `xircuits --branch branchName` to pull `xai-components`
2. `xircuits-examples --branch branchName` to pull `examples` and `datasets`

from your specified branch name. I've also added error handling for file pulling and invalid branches.

## References

https://github.com/XpressAI/xircuits/pull/131 https://github.com/XpressAI/xircuits/pull/132

## Pull Request Type


## Type of Change


# Tests

The branch `fahreza/component-submodules` can be used to test differences between master and branch.

**1. Test Entry Point**
1. Download wheel from latest github action
2. Launch xircuits with `xircuits --branch fahreza/component-submodules`, accept to download `xai_components` when prompted
3. Verify that pycaret and modelstash components do not exist
  
**2. Test Example Downloading**
1. Download examples with `xircuits-examples --branch fahreza/component-submodules`
2. Verify that AutoML and MS xircuits examples do not exist

**3. Test Exception Handling**
1. Try  `xircuits --branch` or `xircuits-examples --branch` with an invalid branch name
2. Verify that it returns `examples from XpressAi/Xircuits branch invalidBranch does not exist!`

## Tested on?


# Notes

1. I've lowercased the `Datasets` directory to match the rest of the repository. 
2. The submodules are treated as `files` instead of `dir` in pygithub and will return a invalid file message. Nonetheless, the rest of the components will be downloaded.
